org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/configuration/GradleArguments.java

```diff
@@ -138,4 +147,32 @@ private static String buildScanArgumentFor(BuildEnvironment environment) {
         }
     }
 
+    private static List<String> getInitScriptArguments() {
+        File initScript = getEclipsePluginInitScriptLocation();
+        try {
+            if (!initScript.exists()) {
+                Files.createParentDirs(initScript);
+                Files.touch(initScript);
+                URL resource = GradleVersion.class.getResource(INITSCRIPT_LOCATION);
+                if (resource == null) {
+                    throw new GradlePluginsRuntimeException(String.format("Resource '%s' not found.", INITSCRIPT_LOCATION));
+                }
+
+                URLConnection connection = resource.openConnection();
+                connection.setUseCaches(false);
```

Comment:
Tbh I'm not sure if it is needed. I was looking at other parts of the code, in this case GradleVersion. I suspected this might be an eclipse-plugin-ism so I copied from there